File: Dart/Structures/Stack.dart
class Node {
  Node? next;
  late int data;
  Node(int data) {
    this.data = data;
    this.next = null;
  }
}

class Stack {
  Node? head;
  late int size;
  Stack() {
    this.size = 0;
    this.head = null;
  }

  void push(int data) {
    Node node = new Node(data);
    if (isEmpty())
      this.head = node;
    else {
      node.next = this.head;
      this.head = node;
    }
    ++this.size;
  }

  void pop() {
    this.head = this.head?.next;
    --this.size;
  }

  void display() {
    Node? ptr = this.head;
    while (ptr?.next != null) {
      print("${ptr?.data} ");
      ptr = ptr?.next;
    }
  }

  int stackSize() {
    return this.size;
  }

  bool isEmpty() {
    return this.head == null;
  }
}

void main() {
  var s = new Stack();
  for (int i = 0; i < 10; ++i) {
    s.push(i);
  }
  s.display();
  s.pop();
  print("popped");
  s.display();
}

File: C#/Structures/Stack.cs
using System;

class Stack{
    public class Node{
        public int data;
        public Node next = null;
        public Node(int data){
            this.data = data;
        }
    }

    Node top = null;
    Node bottom = null;
    private int m_size;

    public Stack(){
        this.m_size = 0;
    }

    public void push(int data){
        Node node = new Node(data);
        node.next = this.top;
        this.top = node;
        ++ this.m_size;
    }

    public void pop(){
        if(is_empty()){
            //error
        }
        else
            this.top = this.top.next;
        -- this.m_size;
    }

    public void display(){
        Node ptr = this.top;
        while(ptr.next != null){
            Console.Write(ptr.data + " ");
            ptr = ptr.next;
        }

    }
    public int top_value(){
        return this.top.data;
    }

    public bool is_empty(){
        return this.top == null && this.m_size == 0;
    }

    public int size(){return this.m_size;}
}

public class MainStackConcrete{
    public static void Main(string[] args){
        Stack s = new Stack();
        
        for(int i = 0; i<10; ++i)
            s.push(i);
        
        s.display();
        Console.WriteLine("popped one value");

        s.pop();
        s.display();
    }
}